x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/RestVectorTileAction.java

```diff
@@ -309,6 +346,18 @@ private static VectorTile.Tile.Layer.Builder buildAggsLayer(
                 }
             }
             aggLayerBuilder.addFeatures(featureBuilder);
+            if (request.getWithLabels()) {
+                // Add label position as point
+                featureBuilder.clear();
+                GeoPoint labelPos = (GeoPoint) bucket.getKey();
+                byte[] labelPosFeature = featureFactory.point(labelPos.lon(), labelPos.lat());
+                if (labelPosFeature != null && labelPosFeature.length != 0) {
+                    featureBuilder.mergeFrom(labelPosFeature);
+                    VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, KEY_TAG, bucketKey);
+                    VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, LABEL_POSITION_TAG, true);
+                    aggLayerBuilder.addFeatures(featureBuilder);
+                }
```

Comment:
I think we agreed that we should add all the tags that we added to the feature that is not a tag. That brings the question to what happens if the user has defined an attribute with the same name as the label?